classify.c

```diff
@@ -604,19 +607,42 @@ static int check_for_irq_ban(int irq, GList *proc_interrupts)
 	return 0;
 }
 
+static void get_mod_name(char *path, char *mod)
+{
+	char *cmd;
+	FILE *output;
+
+	if (!path)
+		return;
+
+	cmd = alloca(strlen(path) + 30);
```

Comment:
why +30 here?

Reply:
It is prepared for the remaining string "cat %s/uevent | grep \"DRIVER=\""

Reply:
yeah, but 30?  Thats just a random sizing of the string.  at least define your template and take its strlen, i.e.:
#define CMDTMPL "/uevent | grep \"DRIVER=\""
...
cmd = alloca(strlen(path) + strlen(CMDTMPL) + 1); //alloc cmd template for path + null terminator byte